squire_sdk/src/server/gathering/hall.rs

```diff
@@ -55,26 +49,26 @@ pub enum GatheringHallMessage {
     NewGathering(TournamentId),
     /// Adds an onlooker to a gathering
     NewConnection(TournamentId, SessionWatcher, WebSocket),
-    /// Perist all the tournaments that need to be persisted
+    ///
+    PersistReady(TournamentId),
+    /// Persist all the tournaments that need to be persisted
     Persist,
+    /// Destroy a gathering when the gathering decides that it should be terminated
+    DestroyGathering(TournamentId, OneshotSender<()>),
```

Comment:
This is very minor, but did you consider using a non-unit, zero-sized type? Basically, a type that is functionally identical to a unit but conveys meaning in its name, say `pub struct Completed`.

Reply:
No I didn't. I am not very familiar with the benefits of that. If there is a resource you can share I'd love to look into it.

Reply:
The only benefit is that it's easier to understand. At runtime, there is no difference between a `struct Completed;` and a `()`. However, seeing `OneshotSender<Completed>` makes it easier to remember why you need a sender at all. I bring this up only because, while reviewing this, it took me a little while to remember the reasoning for why we were sending a unit back.